### ISSUE TITLE
Rehype-prism-plus with @next/mdx & @mdx-js/loader

### ISSUE THREAD

**Issue by ellcrev** [timlrx/rehype-prism-plus#44]
Hi, new to MDX. I am currently using @next/mdx, @mdx-js/loader and @mdx-js/react. My code looks like:

```js
import mdx from "@next/mdx";
import rehypePrism from "rehype-prism-plus";

const withMDX = mdx({
  extension: /\.mdx?$/,
  options: {
    remarkPlugins: [],
    rehypePlugins: [rehypePrism],
  },
});

export default withMDX({
  reactStrictMode: true,
  pageExtensions: ["ts", "tsx", "js", "jsx", "md", "mdx"],
});
```

I keep getting `TypeError: Converting circular structure to JSON
    --> starting at object with constructor 'State'`

Other rehype plugins like rehypeKatex work normally.

Is there some other remark or rehype plugin that needs to be integrated before or after rehype-prism-plus? Thanks.

**Comment by ellcrev** [comment]
Replaced rehype-prism-plus with @mapbox/rehype-prism and that also works normally.

**Comment by zackchengyk** [comment]
Having the same problem. I'm using `@mdx-js/mdx`, `@mdx-js/react`, and `@mdx-js/rollup`.

I suspect the TypeError is a red herring; that comes from a `JSON.stringify` call which is part of the error logging after a failed attempt by `source-map` to validate the source map. The source map being incorrect is probably a sign for the underlying issue.

**Comment by zackchengyk** [comment]
@ellcrev , after making my own "version" for debugging by copying this repository's `src/generators.js` (and using it directly instead of the package import), I've figured out a workaround.

### Problem

It seems the problem is that, for whatever reason, the `position` fields of the `node`s in the output syntax tree are causing _something_ to fail. These fields are set by the `rehype-prism-plus` code, which modifies the input syntax tree named [here](https://github.com/timlrx/rehype-prism-plus/blob/3560f0ddbc74fa8bb9d8918050e905654b841849/src/generator.js#L164). 

I'm not sure why this would be a problem, but it's an easy fix:

### Solution

By stripping the output tree's nodes of their `position` fields, everything seems to work again. Simply call the following function on the `node` [at line 284](https://github.com/timlrx/rehype-prism-plus/blob/3560f0ddbc74fa8bb9d8918050e905654b841849/src/generator.js#L284).

```js
const recursivelyStripPositions = (node) => {
  delete node.position

  if (!node.children || node.children.length === 0) return node

  node.children = node.children.map((x) => recursivelyStripPositions(x))

  return node
}
```

I believe this is a perfectly serviceable solution, as:
1. At least one other `rehype` plugin ([`starry-night`](https://github.com/wooorm/starry-night), see the [instructions on its readme](https://github.com/wooorm/starry-night#example-integrate-with-unified-remark-and-rehype) for how to turn it into a `rehype` plugin) produces nodes which don't even have `position` fields, and2. 
2. The `position` fields seem to have to do with the markdown instead of the final html, so it's not important, and3. 
3. The `position` fields appear to have been incorrect to begin with, anyway. Even though `rehype-prism-plus` was setting the `position.line` fields correctly, it wasn't setting the `position.column` fields to anything other than 0.4. 

There is of course, a "better" "fix": don't set the position in the first place. I just didn't want to dive too deeply into the code.

### Explanation

I'm not qualified to comment on why the above is a problem (perhaps @timlrx could?), given my mere four hours of experience with `unified` and its ecosystem.

**Comment by ellcrev** [comment]
Great fix! Works perfectly now.

**Comment by timlrx** [comment]
Thanks @zackchengyk for looking into it and proposing a workaround. The node positions are required to correctly identify elements that belong to the same line and assign them to the same `div` with an optional line number. `position` field is not required and was just added to satisfy the ts compiler. This position `node` mimics the behaviour of converting it to html and parsing it with `rehype-parse` again which adds positioning info.

`starry-night` or `refractor`, the prism equivalent does not need to do that. You are right that it is not required in the final output and can be removed.

Will check out the code again later in the week and see what's the underlying issue. Thanks for the patience.

**Comment by timlrx** [comment]
Tried it out with other libraries - unified, mdx-bundler and it works fine so I think it might be something specific to @mdx/loader. In any case, rather than diving deeper, I just removed the positions in the final node as suggested by @zackchengyk.

Published the updated version as v1.4.2.

**Comment by timlrx** [comment]
I pushed a new fix which adds the `column` field required by the node position, which seems to work fine with @mdx/loader. This means I can remove `recursivelyStripPositions`. This is published as v1.4.3. Let me know if you run into issues.
